docs/learn/dapp-developer/read-asset-data.md

```diff
@@ -159,33 +159,35 @@ console.log(tokenType);
-
 At this point, you should be able to identify if the digital asset is a:
 
 - LSP7 - Token
 - LSP7 - NFT
 - LSP8 - NFT
 - LSP8 - Collection
 
-Based on this information, the asset metadata is fetched in different ways.
-
 ## Fetch the Asset Metadata
 
-To fetch the whole contracts JSON file, you can use the [`fetchData('LSP4Metadata')`](../../tools/erc725js/classes/ERC725.md#fetchdata) directly on the asset address. Please check on the [LSP4 Digital Asset Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata/#types-of-digital-assets) page where to fetch the metadata. In the case of a collection, you will have to fetch the `LSP4Metadata` for each token address within the collection.
+Based on the [token type](#detect-the-token-type), the information of the [LSP4 Digital Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata.md#types-of-digital-assets) can be interpreted and attached in different ways:
 
-<!-- We might need to make it more clear for each token type how to fetch the data and get the tokenId etc. -->
+- As [**global token information**](#global-token-information) of the contract (Token or LSP7 NFT)
+- To each [**individual token ID**](#token-id-metadata) (LSP8 NFT or Collection)
+
+:::tip Token ID Entries
+
+- If your asset is a Token (`tokenType = 0`), individual metadata **cant be set**, due to the lack of token IDs.
+- If your asset is an NFT (`tokenType = 1`), individual metadata **may be set in addition** to the global token information.
+- If your asset is a Collection (`tokenType = 2`), the token ID-specific medatada is **mandatory**.
+
+:::
+
+### Global Token Information
+
+To fetch the whole JSON file of the asset's metadata, you can simply use the [`fetchData('LSP4Metadata')`](../../tools/erc725js/classes/ERC725.md#fetchdata) method of the [`erc725js`](../../tools/erc725js/getting-started.md) library. Optionally, you can call the [`getData('LSP4Metadata')`](../../contracts/contracts/ERC725/ERC725.md#getdata) function on the asset contract and decode the information manually.
 
 :::info differece between get and fetch
 
-The [`getData(...)`](../../tools/erc725js/classes/ERC725#getdata) function only retrieves the data keys values from the smart contract. In comparison, [`fetchData(...)`](../../tools/erc725js/classes/ERC725#fetchdata) will download and decode the content of `VerifiableURI` referenced within the smart contract.
+The [`getData(...)`](../../tools/erc725js/classes/ERC725#getdata) function only retrieves the data keys values from the smart contract. In comparison, [`fetchData(...)`](../../tools/erc725js/classes/ERC725#fetchdata) will download and decode the content of `VerifiableURI` referenced within the smart contract. You can read more about `VerifiableURI` in the [LSP2 Specification](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#verifiableuri).
```

Comment:
This cam be grouped with above:

--- 

To fetch the whole JSON file of the asset's metadata, you can use one of the following 2 functions from _erc725.js_:
- [`fetchData('LSP4Metadata')`](../../tools/erc725js/classes/ERC725.md#fetchdata): this will download and decode the content of the [`VerifiableURI`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#verifiableuri). as JSON.
- [`getData('LSP4Metadata')`](../../contracts/contracts/ERC725/ERC725.md#getdata): this will retrieve the raw data value from the smart contract. You will then need to decode the `VerifiableURI` manually using `decodeData(...)`.